src/main/java/seedu/address/logic/commands/EditCommand.java

```diff
@@ -305,27 +311,57 @@ public String toString() {
             return super.toString();
         }
     }
+
     /**
      * Stores the details to edit the specialist with. Each non-empty field value will replace the
      * corresponding field value of the specialist.
      */
     public static class EditSpecialistDescriptor extends EditPersonDescriptor {
-        public EditSpecialistDescriptor() {}
-
+        private Specialty specialty;
         /**
          * Copy constructor.
          * A defensive copy of {@code tags} is used internally.
          */
         public EditSpecialistDescriptor(EditSpecialistDescriptor toCopy) {
             super(toCopy);
+            setSpecialty(toCopy.specialty);
+        }
+        public EditSpecialistDescriptor() {}
+        public void setSpecialty(Specialty specialty) {
+            this.specialty = specialty;
         }
+
+        public Optional<Specialty> getSpecialty() {
+            return Optional.ofNullable(specialty);
+        }
+
         @Override
         public boolean equals(Object other) {
-            return super.equals(other);
+            if (super.equals(other) && other instanceof EditSpecialistDescriptor) {
+                EditSpecialistDescriptor otherEditSpecialistDescriptor = (EditSpecialistDescriptor) other;
+                return Objects.equals(specialty, otherEditSpecialistDescriptor.specialty);
+            }
+            return false;
         }
+
         @Override
         public String toString() {
-            return super.toString();
+            return new ToStringBuilder(this)
+                    .add("name", super.getName().orElse(null))
```

Comment:
I think we could reduce some code repetition here by utilizing the superclass toString. Take a look at [the docs](https://commons.apache.org/proper/commons-lang/apidocs/org/apache/commons/lang3/builder/ToStringBuilder.html). There seems to be an `appendSuper` method you could use to resolve this.

Reply:
Agreed it is repetitive, but I think our project is using its own `ToStringBuilder` class, not the one you linked.
I think `appendSuper` isn't appropriate as `toString` returns has a `OBJECT_SUFFIX` at the end, resulting in appended Strings being inconsistent.
I did an alternative solution by adding a appending method in `StringUtil.java` which is more specific to our needs for our current `toString()` format.

Reply:
Yes, my bad on linking the wrong docs. I think the updated method is good! Hope you can resolve the other comments so that the PR can be merged!

Reply:
Yep, agreed, my bad for linking the wrong docs. Good job on the update!